packages/instantsearch.js/src/types/results.ts

```diff
@@ -105,7 +105,7 @@ type InitialResult = {
     params: RecommendParametersOptions['params'];
     results: RecommendResults['_rawResults'];
   };
-  requestParams?: SearchOptions;
+  requestParams?: SearchOptions[];
```

Comment:
should this still be optional? maybe InitialResult as a whole optional but each key required?

Reply:
Could be an empty array maybe, I still think it's undefined when there are no search widgets and only recommend ones though